src/datacite/datacite.py

```diff
@@ -0,0 +1,311 @@
+    session: requests.Session,
+    url: str,
+    params: Dict[str, Any],
+    timeout: int = 60,
+    initial_backoff: int = 2,
+    max_backoff: int = 60,
+) -> requests.Response:
+    """Make HTTP GET request with exponential backoff for retryable errors.
+
+    Automatically retries requests on rate limiting (429) and server errors
+    (500, 502, 503, 504) using exponential backoff strategy.
+
+    Args:
+        session: Requests session object for making HTTP calls.
+        url: URL to send the GET request to.
+        params: Query parameters to include in the request.
+        timeout: Request timeout in seconds. Defaults to 60.
+        initial_backoff: Initial backoff delay in seconds. Defaults to 2.
+        max_backoff: Maximum backoff delay in seconds. Defaults to 60.
+
+    Returns:
+        Successful HTTP response object with status code 200.
+
+    Raises:
+        requests.HTTPError: For non-retryable HTTP errors (e.g., 400, 401, 404).
+    """
+    backoff = initial_backoff
+    while True:
+        resp = session.get(url, params=params, timeout=timeout)
+        if resp.status_code == 200:
+            return resp
+        if resp.status_code in (429, 500, 502, 503, 504):
+            time.sleep(backoff)
+            backoff = min(backoff * 2, max_backoff)
+            continue
+        resp.raise_for_status()
+
+
+def _build_user_agent_headers(user_agent_mailto: Optional[str] = None) -> Dict[str, str]:
+    """Build HTTP headers with User-Agent for DataCite API.
+
+    DataCite provides higher rate limits (1000 requests per 5 minutes) for
+    identified unauthenticated users who include an email in the User-Agent.
+
+    Args:
+        user_agent_mailto: Email address to include in User-Agent header
+            for increased rate limits. If None, no User-Agent is set.
+
+    Returns:
+        Dictionary of HTTP headers. Empty if no email provided, otherwise
+        contains User-Agent header with format:
+        "SageBionetworks-DPE/1.0 (mailto:email@example.com)".
+    """
+    headers = {}
+    # Identified unauthenticated users get 1000 req/5min when User-Agent includes mailto.
+    # Example: "Sage-DPE/1.0 (mailto:phil.snyder@sagebase.org)"
+    if user_agent_mailto:
+        headers["User-Agent"] = f"SageBionetworks-DPE/1.0 (mailto:{user_agent_mailto})"
+    return headers
+
+
+def _should_continue_pagination(data: List[Dict[str, Any]], page_size: int) -> bool:
+    """Determine if pagination should continue based on returned data.
+
+    Pagination stops when either no data is returned or a short page
+    (fewer items than page_size) is received, indicating the last page.
+
+    Args:
+        data: List of data items returned from the current page.
+        page_size: Expected number of items per full page.
+
+    Returns:
+        True if pagination should continue (full page received),
+        False if this is the last page or no data was returned.
+    """
+    return bool(data) and len(data) >= page_size
```

Comment:
Do we expect more items than current page size if we are just doing one page at a time? Is this just to account for API edge case scenario?